### PR TITLE
[code] feat: add linting with pylint

### PR DESCRIPTION
**Pull request by grejdi-mbta**
This PR introduces Pylint to CI for linting python files inside of `py_cubic_ods_ingestion/lib/` and `py_cubic_ods_ingestion/tests/`. Pylint configuration is borrowed from the Google's [Python Style Guide](https://google.github.io/styleguide/pyguide.html) with the only modifications being `max-line-length` and `indent-string`.

### CODE REVIEW

File: py_cubic_ods_ingestion/pyproject.toml [hunk 1]
```diff
@@ -39,7 +40,8 @@ show_error_codes = true
 [[tool.mypy.overrides]]
 module = [
   'pyspark.*',
-  'awsglue.*'
+  'awsglue.*',
+  'py_cubic_ods_ingestion.*'
```
Comment (paulswartz) [inline py_cubic_ods_ingestion/pyproject.toml:44]: question: this looks like it's disabling typechecking for our module? at least partially.

File: py_cubic_ods_ingestion/lib/py_cubic_ods_ingestion/ingest_incoming.py [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
-from pyspark.context import SparkContext
-from awsglue.context import GlueContext
-from awsglue.job import Job
-from awsglue.dynamicframe import DynamicFrame
-from pyspark.sql.functions import lit
+from awsglue.context import GlueContext  # pylint: disable=import-error
+from awsglue.job import Job  # pylint: disable=import-error
+from awsglue.utils import getResolvedOptions  # pylint: disable=import-error
+from pyspark.context import SparkContext  # pylint: disable=import-error
+from pyspark.sql.functions import lit  # pylint: disable=import-error
```
Comment (paulswartz) [inline py_cubic_ods_ingestion/lib/py_cubic_ods_ingestion/ingest_incoming.py:11]: question: I thought you were going to add this as a dev dependency?

File: py_cubic_ods_ingestion/pyproject.toml (hunk 1 above)
Comment (grejdi-mbta) [reply py_cubic_ods_ingestion/pyproject.toml:44]: It wasn't. I was playing around with some settings, so this was left over. I hadn't reviewed the PR yet, so you caught this before I cleaned up. Please wait for a request for review. 😉

File: py_cubic_ods_ingestion/lib/py_cubic_ods_ingestion/ingest_incoming.py (hunk 2 above)
Comment (grejdi-mbta) [reply py_cubic_ods_ingestion/lib/py_cubic_ods_ingestion/ingest_incoming.py:11]: In another PR. I haven't added the Pyspark testing.

File: py_cubic_ods_ingestion/pyproject.toml (hunk 1 above)
Comment (paulswartz) [reply py_cubic_ods_ingestion/pyproject.toml]: My apologies! I get the e-mails about the commits, and sometimes miss that I wasn't a reviewer yet :)

**Review by paulswartz — APPROVED** [review]
:cake:

**Pull request merged**